x/events/events.go

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package events
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/gofrs/uuid"
+	otelattr "go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	f "github.com/ory/kratos/selfservice/flow"
```

Comment:
Hm, AFAIK the convention is to not have any dependencies to other kratos packages within x, as it is meant to be importable everywhere. I would suggest to move this into the flow package instead.

Reply:
Just inline the necessary interface:
```go
NewSessionIssued(ctx context.Context, sessionID, identityID uuid.UUID, flow interface{GetType() string}) (string, trace.EventOption) {
```

Reply:
I didn't know about this convention.

Inlining the interface doesn't work, because `flow.Flow.GetType()` returns `flow.Type` and not `string`.
I could instead just take the raw string instead of the whole `Flow` object or remove the attribute completely (YAGNI?).

WDYT?

Reply:
YAGNI that shit.